lsp-pyls.el

```diff
@@ -353,53 +359,77 @@ parameters referenced in config."
+           ("pyls.plugins.flake8.enabled" lsp-pyls-plugins-flake8-enabled)
+           ("pyls.plugins.flake8.exclude" lsp-pyls-plugins-flake8-exclude)
+           ("pyls.plugins.flake8.filename" lsp-pyls-plugins-flake8-filename)
+           ("pyls.plugins.flake8.hangClosing" lsp-pyls-plugins-flake8-hang-closing)
+           ("pyls.plugins.flake8.ignore" lsp-pyls-plugins-flake8-ignore)
+           ("pyls.plugins.flake8.maxLineLength" lsp-pyls-plugins-flake8-max-line-length)
+           ("pyls.plugins.flake8.select" lsp-pyls-plugins-flake8-select)
+           ("pyls.plugins.flake8.config" lsp-pyls-plugins-flake8-config)
+           ("pyls.plugins.preload.modules" lsp-pyls-plugins-preload-modules)
+           ("pyls.plugins.preload.enabled" lsp-pyls-plugins-preload-enabled t)
+           ("pyls.plugins.mccabe.threshold" lsp-pyls-plugins-mccabe-threshold)
+           ("pyls.plugins.mccabe.enabled" lsp-pyls-plugins-mccabe-enabled t)
+           ("pyls.plugins.jedi_symbols.all_scopes" lsp-pyls-plugins-jedi-symbols-all-scopes t)
+           ("pyls.plugins.jedi_symbols.enabled" lsp-pyls-plugins-jedi-symbols-enabled t)
+           ("pyls.plugins.jedi_signature_help.enabled" lsp-pyls-plugins-jedi-signature-help-enabled t)
+           ("pyls.plugins.jedi_references.enabled" lsp-pyls-plugins-jedi-references-enabled t)
+           ("pyls.plugins.jedi_hover.enabled" lsp-pyls-plugins-jedi-hover-enabled t)
+           ("pyls.plugins.jedi_definition.follow_builtin_imports" lsp-pyls-plugins-jedi-definition-follow-builtin-imports t)
+           ("pyls.plugins.jedi_definition.follow_imports" lsp-pyls-plugins-jedi-definition-follow-imports t)
+           ("pyls.plugins.jedi_definition.enabled" lsp-pyls-plugins-jedi-definition-enabled t)
+           ("pyls.plugins.jedi_completion.include_params" lsp-pyls-plugins-jedi-completion-include-params t)
+           ("pyls.plugins.jedi_completion.enabled" lsp-pyls-plugins-jedi-completion-enabled t)
+           ("pyls.configurationSources" lsp-pyls-configuration-sources))
+         (when lsp-pyls-plugins-jedi-use-pyenv-environment
+           (let ((pyenv-get-python-executable
+                  (lambda ()
+                    (let* ((pyenv-version (getenv "PYENV_VERSION"))
+                           (root (nth 0 (lsp-find-roots-for-workspace lsp--cur-workspace (lsp-session)))))
+                      (when root
+                        (setenv "PYENV_VERSION" nil)
+                        (let ((python-env (f-parent
+                                           (f-parent
+                                            (shell-command-to-string
+                                             (format "PYENV_DIR='%s' %s which python"
+                                                     root lsp-pyls-pyenv-command-path))))))
+                          (message (format "Configure pyls with environment: %s" python-env))
+                          (setenv "PYENV_VERSION" pyenv-version)
+                          python-env)
```

Comment:
nit: as per elisp coding guidelines(which we mostly follow) the brackets must be closed on the same line.